sycl/doc/extensions/supported/sycl_ext_intel_dataflow_pipes.asciidoc

```diff
@@ -642,55 +642,70 @@ Automated mechanisms are possible to provide uniquification across calls, and co
+* `sycl::ext::oneapi::experimental::latency_anchor_id<N>`, where `N` is an integer: An ID to associate with the current read/write function call, which can then be referenced by other `latency_constraint` properties elsewhere in the program to define relative latency constaints. ID must be unique within the application, and a diagnostic is required if that condition is not met.
+* `sycl::ext::oneapi::experimental::latency_constraint<A, B, C>`: A tuple of three values which cause the current read/write function call to act as an endpoint of a latency constraint relative to a specified `latency_anchor_id` defined by a different instruction.
 ** `A` is an integer: The ID of the target anchor defined on a different instruction through a `latency_anchor_id` property.
-** `B` is an enum value: The type of control from the set {`type::exact`, `type::max`, `type::min`}.
+** `B` is an enum value: The type of control from the set {`latency_control_type::exact`, `latency_control_type::max`, `latency_control_type::min`}.
 ** `C` is an integer: The relative clock cycle difference between the target anchor and the current function call, that the constraint should infer subject to the type of the control (exact, max, min).
 
-The template arguments above don't have to be specified if user doesn't want to apply latency controls. The template arguments can be passed in arbitrary order.
-
-=== Implementation
+=== Synopsis
 
 [source,c++]
 ----
 // Added in version 2 of this extension.
-namespace sycl::ext::intel::experimental {
-enum class type {
-  none, // default
-  exact,
-  max,
-  min
-};
-
-template <int32_t _N> struct latency_anchor_id {
-  static constexpr int32_t value = _N;
-  static constexpr int32_t default_value = -1;
-};
-
-template <int32_t _N1, type _N2, int32_t _N3> struct latency_constraint {
-  static constexpr std::tuple<int32_t, type, int32_t> value = {_N1, _N2, _N3};
-  static constexpr std::tuple<int32_t, type, int32_t> default_value = {
-      0, type::none, 0};
-};
+#include <sycl/ext/oneapi/latency_control/properties.hpp>
+#include <sycl/ext/oneapi/properties/properties.hpp>
+//***********sycl/ext/oneapi/latency_control/properties.hpp***************
+// enum class latency_control_type {
+//   none, // default
+//   exact,
+//   max,
+//   min
+// };
+//
+// struct latency_anchor_id_key {
+//   template <int Anchor>
+//   using value_t = property_value<latency_anchor_id_key,
+//                                  std::integral_constant<int, Anchor>>;
+// };
+//
+// struct latency_constraint_key {
+//   template <int Target, latency_control_type Type, int Cycle>
+//   using value_t =
+//       property_value<latency_constraint_key,
+//                      std::integral_constant<int, Target>,
+//                      std::integral_constant<latency_control_type, Type>,
+//                      std::integral_constant<int, Cycle>>;
+// };
+//************************************************************************
```

Comment:
Why is the synopsis above commented out?  The definition of `pipe` below isn't commented out.

Normally, we just list all of the public APIs in the synopsis without any indication of the header file that defines the API because normally all extensions are provided by `<CL/sycl.hpp>` / `<sycl/sycl.hpp>`.  Does the user need to specifically `#include` a different header in order to get these APIs?  If that is the case, this should be noted in the text part of the specification someplace.  If it is not necessary to include a special header, I'd remove all these `#include` statements from the synopsis.

Normally, properties also have a namespace-scope variable that the application can use to add the property to a property list.  For example:

```
struct latency_anchor_id_key {
  template <int Anchor>
  using value_t = property_value<latency_anchor_id_key, std::integral_constant<int, Anchor>>;
};

template<int Anchor>
inline constexpr latency_anchor_id_key::value_t<Anchor> latency_anchor_id;
```

I assume you have these too?  If so, they should be in the synopsis.

Reply:
Ah now I see the expectation of the synopsis section. I will update it.
My new APIs are provided with <sycl/ext/intel/fpga_extensions.hpp> which has already been stated in the docs so I think we are good.

By the way, can we assume oneAPI users will always include <CL/sycl.hpp> so I don't need to include redundant header files in <sycl/ext/intel/fpga_extensions.hpp>?

Reply:
`<CL/sycl.hp>` doesn't include `<sycl/ext/intel/fpga_extensions.hp>`, but `<CL/sycl.hp>` and `<sycl/ext/intel/experimental/fpga_utils.hpp>` both include `<sycl/ext/oneapi/properties/properties.hpp>`. So I will remove this include from `<sycl/ext/intel/experimental/fpga_utils.hpp>`.

Reply:
> My new APIs are provided with <sycl/ext/intel/fpga_extensions.hpp> which has already been stated in the docs so I think we are good.

I see it in "sycl_ext_intel_fpga_lsu.md", but it looks like it is missing from "sycl_ext_intel_dataflow_pipes.asciidoc".

Reply:
Hmm, good catch. I think the reason "sycl_ext_intel_dataflow_pipes.asciidoc" doesn't mention <sycl/ext/intel/fpga_extensions.hpp> is that this pipe extension is not FPGA-only.

I will add something like "This latency control API is included in `sycl/ext/intel/fpga_extensions.hpp`" .